src/executingstreamidguard.hpp

```diff
@@ -15,16 +15,21 @@
 //*****************************************************************************
 #pragma once
 
+#include "model_metric_reporter.hpp"
 #include "ovinferrequestsqueue.hpp"
 
 namespace ovms {
 
 struct ExecutingStreamIdGuard {
-    ExecutingStreamIdGuard(ovms::OVInferRequestsQueue& inferRequestsQueue) :
+    ExecutingStreamIdGuard(ovms::OVInferRequestsQueue& inferRequestsQueue, ModelMetricReporter& reporter) :
         inferRequestsQueue_(inferRequestsQueue),
         id_(inferRequestsQueue_.getIdleStream().get()),
-        inferRequest(inferRequestsQueue.getInferRequest(id_)) {}
+        inferRequest(inferRequestsQueue.getInferRequest(id_)),
+        reporter(reporter) {
+        INCREMENT_IF_ENABLED(reporter.inferReqActive);
+    }
     ~ExecutingStreamIdGuard() {
+        DECREMENT_IF_ENABLED(reporter.inferReqActive);  // TODO: Order is important?
```

Comment:
stream id actually can be never active if we have queue with futures. Upon returning of the old inferReq it does not get returned to the queue but is reserved for nodestreamidguard. Question is how we want to track this - ie if such infer req is active or no. In my opinion it is not. We may modify queue API to give back such info.

In returnStream in case of promises.size().